mlir/lib/Dialect/Rock/Tuning/RockTuningImpl.cpp

```diff
@@ -658,40 +658,61 @@ LogicalResult getTuningProblemStr(rock::RockGemmWrapperInterface gemmIF,
     for (unsigned i = 0; i < size; ++i) {
       auto filterAttr =
           filterLayoutAttr.getValue()[i].template cast<StringAttr>();
-      fLayoutMap[filterAttr.getValue()] = i;
-    }
-    for (unsigned i = 0; i < size; ++i) {
+      auto fs = filterAttr.getValue();
+      if (fs == "y")
+        fs = "0";
+      if (fs == "x")
+        fs = "1";
+      fLayoutMap[fs] = i;
       auto inputAttr =
           inputLayoutAttr.getValue()[i].template cast<StringAttr>();
-      iLayoutMap[inputAttr.getValue()] = i;
-    }
-    for (unsigned i = 0; i < size; ++i) {
+      auto is = inputAttr.getValue();
+      if (is == "hi")
+        is = "0i";
+      if (is == "wi")
+        is = "1i";
+      iLayoutMap[is] = i;
       auto outputAttr =
           outputLayoutAttr.getValue()[i].template cast<StringAttr>();
-      oLayoutMap[outputAttr.getValue()] = i;
+      auto os = outputAttr.getValue();
+      if (os == "ho")
+        os = "0o";
+      if (os == "wo")
+        os = "1o";
+      oLayoutMap[os] = i;
     }
 
-    SmallString<5> fLayout("#####");
-    SmallString<5> iLayout("#####");
-    SmallString<5> oLayout("#####");
+    SmallString<6> fLayout;
+    SmallString<6> iLayout;
+    SmallString<6> oLayout;
+    fLayout.assign(size, '#');
```

Comment:
I'm pretty sure there's a constructor for this? If there isn't, just resolve this

Reply:
SmallVector has that constructor, but SmallString doesn't.  Go figure.